cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -134,13 +134,18 @@ public synchronized void create(NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().enforce(instanceId, principal, Action.ADMIN);
     }
 
+    // store the meta first in the namespace store because namespacedlocationfactory need to look up location
+    // mapping from namespace config
```

Comment:
NamespaceAdmin knows if there is ordering or not if the NamespacedLocationFactory does not determine this on its own then this information need to be passed down to it. The current get of NamespacedLocationFactory does not have a parameter to say this if we change it  that means all the classes calling it will need to do the lookup on their own and pass down whether there is custom mapping or not. This seems weird since this should be encapsulated inside the NamespacedLocationFactory. Unless we add another method like create in namespaced location factory which is used only by the NamespaceAdmin while creating the namespaceadmin which does not do lookup on its own but depend on the called to tell if there is mapping or not. Should we do that ?